plugins/search.cpp

```diff
@@ -249,7 +252,7 @@ class search_generic
                 // ENTER or ESC: leave typing mode
                 end_entry_mode();
             }
-            else if (cursor_key_pressed(input))
+            else if (cursor_key_pressed(input, entry_mode))
```

Comment:
On second glance, I think this call isn't technically correct - most screens (other than fortress mode) use STANDARDSCROLL keys to scroll (or maybe even SECONDSCROLL), but cursor_key_pressed checks for CURSOR keys. Worth addressing, but out of scope here, and it works with default keybindings.